components/led_strip/led_vu.c

```diff
@@ -0,0 +1,341 @@
+/* 
+ *  Control of LED strip within squeezelite-esp32 
+ *     
+ *  (c) Wizmo 2021
+ *
+ *  Loosely based on code by 
+ *     Chuck Rohs 2020, chuck@zethus.ca
+ * 
+ *  This software is released under the MIT License.
+ *  https://opensource.org/licenses/MIT
+ *
+ * ToDo:
+ * Driver does support other led device. Maybe look at supporting in future. 
+ * Bright supports up to 255, but expect bad behaviour above 128.
+ * The VU refresh rate has been decreaced (100->75) to optimize animation of spin dial.  Could make
+ *   configurable like text scrolling (or use the same value)  
+ * Artwork function, but not released as very buggy and not really practical
+ */
+
+#include <ctype.h>
+#include <math.h>
+#include "esp_log.h"
+
+#include "led_strip.h"
+#include "platform_config.h"
+#include "led_vu.h"
+
+static const char *TAG = "led_vu";
+
+#define LED_VU_STACK_SIZE 	(3*1024)
+#define LED_VU_RMT_INTR_NUM 20U
+
+#define LED_VU_PEAK_HOLD 6U
+
+#define max(a,b) (((a) > (b)) ? (a) : (b))
+
+static struct led_strip_t* led_vu = NULL;
+static struct led_strip_t  led_strip_config = {
+    .rgb_led_type      = RGB_LED_TYPE_WS2812,
+    .rmt_channel       = RMT_CHANNEL_1,
+    .rmt_interrupt_num = LED_VU_RMT_INTR_NUM,
+    .gpio              = GPIO_NUM_22,
+};
+
+static struct {
+    int gpio;
+    int length;
+    uint8_t bright;
+    int vu_length;
+    int vu_start_l;
+    int vu_start_r;
+    int vu_odd;
+} strip = { .gpio = 22, .length = 19, .bright = 20 };
```

Comment:
You should not have hard-coded values. Use sdkconfig settings with optionally the "LOCKED" option (see led.c or monitor.c)